### DISCUSSION TITLE
How to implement multiple-to-multiple linkage in formliy v2?

### DISCUSSION THREAD

**Discussion by RihuaChen** [alibaba/formily#1377] [Q&A 2.x] [answered]
We can use rxjs operator 'combineLatest' in formliy 1.0, but I could not found the same solution in V2 version, could you help me with this issue?

**Comment by janryWang** [comment, marked as answer]
Many-to-many can actually be converted into one-to-many
https://v2.formilyjs.org/guide/advanced/linkages#one-to-many-linkage

**Reply by RihuaChen** [reply]
I have a little confuse about this one, for example combineLatest([observer1, observer2]).subscribe(([v1, v2]) => { }), in this case we can get v1, v2 at same time, but I don't know how to get these v1 & v2 at same time in one-to-many demo.

- By the way, the requirement from my side is that I need to do something when v1 === 'someVal' && v2 === 'otherVal', just like this scenario.

**Reply by janryWang** [reply]
effect style
```
onFieldReact('aa',()=>{
   if(form.values.v1 === something1){
     //...
   }
   if(form.values.v2 === something2){
    //...
   }
})
```
reactions style
```
<Field name="aa"  component={[Input]} reactions={(field,form)=>{
   if(form.values.v1 === something1){
     //...
   }
   if(form.values.v2 === something2){
    //...
   }

}} />
```

**Reply by RihuaChen** [reply]
Thanks for your patient help, is that means if I want to change the value of  both 'aa' and 'bb' at same time, I need to write the logic both in 'aa' field and 'bb' field ? Or I can use

```
onFieldReact('*(aa,bb)',()=>{
   if(form.values.v1 === something1){
     //...
   }
   if(form.values.v2 === something2){
    //...
   }
})
```

**Reply by janryWang** [reply]
yes, that's right

**Reply by RihuaChen** [reply]
Sorry to bother you again, could you help me with these situation, I've create a codeSandbox [https://codesandbox.io/s/heuristic-waterfall-th98k?file=/App.tsx](https://codesandbox.io/s/heuristic-waterfall-th98k?file=/App.tsx)

**Reply by janryWang** [reply]
Here you are! https://codesandbox.io/s/delicate-bash-kp246?file=/App.tsx
When formily has reactive, we can fly

**Reply by RihuaChen** [reply]
Wow, it's crazy ! Big thanks to you !!!
